### DISCUSSION TITLE
Mosaic of stac catalog assets

### DISCUSSION THREAD

**Discussion by charalamm** [developmentseed/titiler#658] [Q&A]
Hello,

Again, thanks for this awesome repo 🥇 

I want to show a mosaic of a very big amount of scenes on a web viewer. I used the mosaic endpoints of titiler and it worked very good! At the same time I am thinking to create a STAC catalog for my data.

So, I was wondering if it is possible to have a mosaic in the viewer using the STAC catalog instead of creating a MosaicJSON. Currently I see that the stac endpoints can only work with single STAC item urls. Is there a (side)way to do achieve showing mosaics by using the STAC catalog in titiler?

Thanks!

**Comment by vincentsarago** [comment]
Hi @charalamm, have you looked at https://github.com/stac-utils/titiler-pgstac 

>  Is there a (side)way to do achieve showing mosaics by using the STAC catalog in titiler?

There isn't real way to pass a `stac` catalog to the tiler itself but that could be possible use a custom Tiler. BUT this means that on each tile request the tiler will have to loop through each Items and see if they intersect with the tile boundary. This might be ok for small catalog but still it will be a lot of process. 

useful links:
- https://developmentseed.org/cogeo-mosaic/advanced/dynamic/

**Reply by charalamm** [reply]
Hello @vincentsarago, thank you for the immediate response!

I just looked a bit in the resources you provided. So, as far as I understand it all solutions that have pgstac as a component need to have the STAC catalog data in the database. I mean, I can not use an external STAC catalog as I would with titiler. Is that right?

(This is my last question since the questions are now more related to another repo)

**Reply by vincentsarago** [reply]
> . I mean, I can not use an external STAC catalog as I would with titiler. Is that right?

It depends, there is a difference between STAC catalog (the file) and STAC API (search)

There is a way to using STAC API in mosaic backend (https://developmentseed.org/cogeo-mosaic/advanced/backends/#stac-backend) but this is not supported by default because the STACBackend needs custom inputs 

> (This is my last question since the questions are now more related to another repo)

It's kinda related 😄 so you can continue here!
